src/connectors/pagination/__tests__/connectPagination-test.js

```diff
@@ -361,113 +380,76 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/pagination/
     });
   });
 
-  describe('routing', () => {
-    const getInitializedWidget = () => {
-      const renderFn = jest.fn();
-      const makeWidget = connectPagination(renderFn);
-      const widget = makeWidget({});
-
-      const helper = algoliasearchHelper({}, '');
-      helper.search = jest.fn();
-
-      widget.init({
+  describe('getWidgetState', () => {
+    test('should give back the object unmodified if the default value is selected', () => {
+      const [widget, helper] = getInitializedWidget();
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
         helper,
-        state: helper.state,
-        createURL: () => '#',
       });
 
-      const { refine } = renderFn.mock.calls[0][0];
+      expect(uiStateAfter).toBe(uiStateBefore);
+    });
 
-      return [widget, helper, refine];
-    };
+    test('should add an entry equal to the refinement', () => {
+      const [widget, helper, refine] = getInitializedWidget();
+      refine(4);
+      const uiStateBefore = {};
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
+      });
 
-    describe('getWidgetState', () => {
-      test('should give back the object unmodified if the default value is selected', () => {
-        const [widget, helper] = getInitializedWidget();
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
-          searchParameters: helper.state,
-          helper,
-        });
+      expect(uiStateAfter).toMatchSnapshot();
+    });
 
-        expect(uiStateAfter).toBe(uiStateBefore);
+    test('should give back the object unmodified if refinements are already set', () => {
+      const [widget, helper, refine] = getInitializedWidget();
+      refine(4);
+      const uiStateBefore = {
+        page: 5,
+      };
+      const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+        searchParameters: helper.state,
+        helper,
       });
 
-      test('should add an entry equal to the refinement', () => {
-        const [widget, helper, refine] = getInitializedWidget();
-        refine(4);
-        const uiStateBefore = {};
-        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
-          searchParameters: helper.state,
-          helper,
-        });
+      expect(uiStateAfter).toBe(uiStateBefore);
```

Comment:
Same comment regarding the value instead of the reference.

Reply:
https://github.com/algolia/instantsearch.js/pull/4004#discussion_r309580941